server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -968,11 +965,14 @@ public Metadata.Builder writeTo(CheckedFunction<IndexMetadata, MapperService, IO
             if (indexMetadata == null) {
                 throw new UnsupportedOperationException("Cannot create index via DocTableInfo.writeTo");
             }
-            MapperService mapperService = createMapperService.apply(indexMetadata);
-            DocumentMapper mapper = mapperService.merge(mapping, MapperService.MergeReason.MAPPING_UPDATE);
+
+            long allowedTotalColumns = MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.get(indexMetadata.getSettings());
```

Comment:
Is it worth moving this setting into DocTableInfo?

Reply:
Eventually yes. My plan was to move it once the existing uses within the MapperService are also gone